src/stat.js

```diff
@@ -127,3 +127,111 @@ export function varianceAll(matrix, unbiased, mean) {
     return (sum2 - (sum1 * sum1) / size) / size;
   }
 }
+
+
+/**
+ * Calculate standard deviation column by column of a matrix
+ * @param {Matrix} matrix - Column index
+ * @return {Array}
+ */
+export function stdDeviationByColumn(matrix) {
+  let stdDeviationByColumn = [];
+  let sum = 0;
+  let mean = [];
+  for (let j = 0; j < matrix.columns; j++) {
+    for (let i = 0; i < matrix.rows; i++) {
+      sum += matrix.get(i, j);
+    }
+    mean[j] = (sum / matrix.rows);
+    sum = 0;
+  }
+  for (let j = 0; j < matrix.columns; j++) {
+    for (let i = 0; i < matrix.rows; i++) {
+      sum += Math.pow(matrix.get(i, j) - mean[j], 2);
+    }
+    stdDeviationByColumn[j] = Math.sqrt(sum / (matrix.rows - 1));
+    sum = 0;
+  }
+  return stdDeviationByColumn;
+}
+
+/**
+ * Calculate the covariance of two matrix
+ * @param {Matrix} xMatrix
+ * @param {Matrix} yMatrix
+ * @param {object} [options]
+ * @param {boolean} [options.center]
+ * @return {Array}
+ */
+export function cov(xMatrix, yMatrix, options = {}) {
+  if (yMatrix === undefined) {
+    yMatrix = xMatrix;
+  }
+  if (xMatrix.rows !== yMatrix.rows) {
+    throw new TypeError('Both matrix must have the same number of rows');
+  }
+  if (options.center === undefined) {
+    options.center = true;
+  }
+  if (options.center === true) {
+    xMatrix.centerColumns();
+    yMatrix.centerColumns();
+  }
+  let covariance = xMatrix.transpose().mmul(yMatrix);
+  for (let i = 0; i < covariance.rows; i++) {
+    for (let j = 0; j < covariance.columns; j++) {
+      let a = covariance.get(i, j) * (1 / (xMatrix.rows - 1));
+      covariance.set(i, j, a);
+    }
+  }
+  return covariance;
+}
+
+/**
+ * Calculate the correlation of two matrix
+ * @param {Matrix} xMatrix
+ * @param {Matrix} yMatrix
+ * @param {object} [options]
+ * @param {boolean} [options.center]
+ * @param {boolean} [optiens.scale]
+ * @return {Array}
+ */
+export function cor(xMatrix, yMatrix, options = {}) {
+  if (yMatrix === undefined) {
+    yMatrix = xMatrix;
+  }
+  if (xMatrix.rows !== yMatrix.rows) {
+    throw new TypeError('Both matrix must have the same number of rows');
+  }
+  if (options.center === undefined) {
+    options.center = true;
+  }
+  if (options.scale === undefined) {
+    options.scale = true;
+  }
+  if (options.center === true && options.scale === true) {
+    xMatrix.centerColumns()
+      .scaleColumns();
```

Comment:
I think it's not the method you want to use. `scaleColumns` does a rescaling to put the numbers between 0 and 1. You probably need to implement a new method that uses the standard deviation.
I'd call it `scale(by, options)`, similar to `center(by)`